File: selector/src/main/java/com/luck/picture/lib/utils/DateUtils.java
package com.luck.picture.lib.utils;

import android.annotation.SuppressLint;
import android.content.Context;

import com.luck.picture.lib.R;

import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.Date;
import java.util.Locale;

/**
 * @author：luck
 * @date：2017-5-25 23:30
 * @describe：DateUtils
 */

public class DateUtils {
    @SuppressLint("SimpleDateFormat")
    private static final SimpleDateFormat SF = new SimpleDateFormat("yyyyMMddHHmmssSSS");
    @SuppressLint("SimpleDateFormat")
    private static final SimpleDateFormat SDF = new SimpleDateFormat("yyyy-MM");

    @SuppressLint("SimpleDateFormat")
    private static final SimpleDateFormat SDF_YEAR = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");

    public static long getCurrentTimeMillis() {
        String timeToString = ValueOf.toString(System.currentTimeMillis());
        return ValueOf.toLong(timeToString.length() > 10 ? timeToString.substring(0, 10) : timeToString);
    }


    public static String getDataFormat(Context context, long time) {
        time = String.valueOf(time).length() > 10 ? time : time * 1000;
        if (isThisWeek(time)) {
            return context.getString(R.string.ps_current_week);
        } else if (isThisMonth(time)) {
            return context.getString(R.string.ps_current_month);
        } else {
            return SDF.format(time);
        }
    }

    public static String getYearDataFormat(long time) {
        time = String.valueOf(time).length() > 10 ? time : time * 1000;
        return SDF_YEAR.format(time);
    }

    private static boolean isThisWeek(long time) {
        Calendar calendar = Calendar.getInstance();
        int currentWeek = calendar.get(Calendar.WEEK_OF_YEAR);
        calendar.setTime(new Date(time));
        int paramWeek = calendar.get(Calendar.WEEK_OF_YEAR);
        return paramWeek == currentWeek;
    }

    public static boolean isThisMonth(long time) {
        Date date = new Date(time);
        String param = SDF.format(date);
        String now = SDF.format(new Date());
        return param.equals(now);
    }


    /**
     * millisecondToSecond
     *
     * @param duration millisecond
     * @return
     */
    public static long millisecondToSecond(long duration) {
        return (duration / 1000) * 1000;
    }

    /**
     * 判断两个时间戳相差多少秒
     *
     * @param d
     * @return
     */
    public static int dateDiffer(long d) {
        try {
            long l1 = getCurrentTimeMillis();
            long interval = l1 - d;
            return (int) Math.abs(interval);
        } catch (Exception e) {
            e.printStackTrace();
            return -1;
        }
    }

    /**
     * 时间戳转换成时间格式
     *
     * @param timeMs
     * @return
     */
    public static String formatDurationTime(long timeMs) {
        String prefix = timeMs < 0 ? "-" : "";
        timeMs = Math.abs(timeMs);
        long totalSeconds = timeMs / 1000;
        long seconds = totalSeconds % 60;
        long minutes = (totalSeconds / 60) % 60;
        long hours = totalSeconds / 3600;
        return hours > 0
                ? String.format(Locale.getDefault(), "%s%d:%02d:%02d", prefix, hours, minutes, seconds)
                : String.format(Locale.getDefault(), "%s%02d:%02d", prefix, minutes, seconds);
    }


    /**
     * 根据时间戳创建文件名
     *
     * @param prefix 前缀名
     * @return
     */
    public static String getCreateFileName(String prefix) {
        long millis = System.currentTimeMillis();
        return prefix + SF.format(millis);
    }

    /**
     * 根据时间戳创建文件名
     *
     * @return
     */
    public static String getCreateFileName() {
        long millis = System.currentTimeMillis();
        return SF.format(millis);
    }

    /**
     * 计算两个时间间隔
     *
     * @param sTime
     * @param eTime
     * @return
     */
    public static String cdTime(long sTime, long eTime) {
        long diff = eTime - sTime;
        return diff > 1000 ? diff / 1000 + "秒" : diff + "毫秒";
    }

}

File: selector/src/main/java/com/luck/picture/lib/utils/DateUtils.kt
package com.luck.picture.lib.utils

import android.content.Context
import com.luck.picture.lib.R
import java.text.SimpleDateFormat
import java.util.*
import kotlin.math.abs

/**
 * @author：luck
 * @date：2022/11/30 3:33 下午
 * @describe：DateUtils
 */
object DateUtils {
    val sf = SimpleDateFormat("yyyyMMddHHmmssSSS", Locale.ENGLISH)
    private val sdfYear = SimpleDateFormat("yyyy-MM-dd HH:mm:ss", Locale.ENGLISH)
    private val sfYm = SimpleDateFormat("yyyy-MM", Locale.ENGLISH)

    /**
     * 时间戳转换成时间格式
     *
     * @param timeMs
     */
    fun formatDurationTime(timeMs: Long): String {
        return formatDurationTime(timeMs, true)
    }

    /**
     * 时间戳转换成时间格式
     *
     * @param timeMs
     */
    fun formatDurationTime(timeMs: Long, isAdjust: Boolean): String {
        if (isAdjust) {
            if (timeMs < 1000) {
                return String.format(
                    Locale.getDefault(), "%s%02d:%02d", "", 0, 1
                )
            }
        }
        val prefix = if (timeMs < 0) "-" else ""
        val absTimeMs = abs(timeMs)
        val totalSeconds = absTimeMs / 1000
        val seconds = totalSeconds % 60
        val minutes = totalSeconds / 60 % 60
        val hours = totalSeconds / 3600
        return if (hours > 0) String.format(
            Locale.getDefault(),
            "%s%d:%02d:%02d",
            prefix,
            hours,
            minutes,
            seconds
        ) else String.format(
            Locale.getDefault(), "%s%02d:%02d", prefix, minutes, seconds
        )
    }

    fun getYearDataFormat(time: Long): String {
        val newTime = if (time.toString().length > 10) time else time * 1000
        return sdfYear.format(newTime)
    }

    fun getDataFormat(context: Context, time: Long): String? {
        val newTime = if (time.toString().length > 10) time else time * 1000
        return when {
            isThisWeek(newTime) -> {
                context.getString(R.string.ps_current_week)
            }
            isThisMonth(newTime) -> {
                context.getString(R.string.ps_current_month)
            }
            else -> {
                sfYm.format(newTime)
            }
        }
    }

    private fun isThisWeek(time: Long): Boolean {
        val calendar = Calendar.getInstance()
        val currentWeek = calendar[Calendar.WEEK_OF_YEAR]
        calendar.time = Date(time)
        val paramWeek = calendar[Calendar.WEEK_OF_YEAR]
        return paramWeek == currentWeek
    }

    private fun isThisMonth(time: Long): Boolean {
        val date = Date(time)
        val param: String = sfYm.format(date)
        val now: String = sfYm.format(Date())
        return param == now
    }

    fun dateDiffer(duration: Long): Boolean {
        return abs(getCurrentTimeMillis() - duration).toInt() <= 1
    }

    private fun getCurrentTimeMillis(): Long {
        val timeToString = System.currentTimeMillis().toString()
        return if (timeToString.length > 10) timeToString.substring(0, 10)
            .toLong() else timeToString.toLong()
    }
}